ciphey/basemods/Checkers/quadgrams.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import re
+from math import log10
+from typing import Dict, Optional
+
+from ciphey.iface import Checker, Config, ParamSpec, T, Translation, registry
+from rich.logging import RichHandler
+
+
+@registry.register
+class Quadgrams(Checker[str]):
+
+    """
+    Uses Quadgrams to determine plaintext
+    """
+
+    def check(self, ctext: T) -> Optional[str]:
+        logging.debug("Trying Quadgrams checker")
+        ctext = re.sub("[^A-Z]", "", ctext.upper())
+        quadgrams = self.QUADGRAMS_DICT
+        N = sum(quadgrams.values())
```

Comment:
Is N == the quadgrams_value? Would it be better renamed?

Reply:
`N` is the sum of all the scores of each quadgram. Each quadgram has a score based on how many times it showed up in a large corpus. So the higher the score of a specific quadgram, the more common it is in English. For example `THAT` is a very common quadgram but `AACX` isn't obviously. I'll rename it to something more descriptive.

Reply:
This sounds like it'd do amazing in docs, nice explanation!